e2e/deployment.go

```diff
@@ -296,3 +297,151 @@ func (rnr *rookNFSResource) Do(action kubectlAction) error {
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error update scale deployment %s: %w", deploymentName, upsErr)
+		}
+		if scaleResult.Spec.Replicas != scale.Spec.Replicas {
+			e2elog.Logf("scale result not matching for deployment %s, desired scale %d, got %d",
+				deploymentName, scale.Spec.Replicas, scaleResult.Spec.Replicas)
+
+			return false, fmt.Errorf("error scale not matching in deployment %s: %w", deploymentName, upsErr)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed update scale deployment %s: %w", deploymentName, err)
+	}
+
+	return nil
+}
+
+func waitForDeploymentUpdate(
+	c kubernetes.Interface,
+	ns,
+	deploymentName string,
+	deployment *appsv1.Deployment,
+	timeout int) error {
+	t := time.Duration(timeout) * time.Minute
+	start := time.Now()
+	err := wait.PollImmediate(poll, t, func() (bool, error) {
+		_, upErr := c.AppsV1().Deployments(ns).Update(context.TODO(), deployment, metav1.UpdateOptions{})
+		if upErr != nil {
+			if isRetryableAPIError(upErr) {
+				return false, nil
+			}
+			e2elog.Logf(
+				"Deployment Update %s/%s has not completed yet (%d seconds elapsed)",
+				ns, deploymentName, int(time.Since(start).Seconds()))
+
+			return false, fmt.Errorf("error updating deployment %s: %w", deploymentName, upErr)
+		}
+
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("failed update deployment %s: %w", deploymentName, err)
+	}
+
+	return nil
+}
+
+func waitForClusterNameUpdate(c kubernetes.Interface, ns, deploymentName string, timeout int) error {
+	e2elog.Logf("waiting for deployment updates %s/%s", ns, deploymentName)
+
+	// Scale down to 0.
+	scale, err := c.AppsV1().Deployments(ns).GetScale(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error get scale deployment %s: %w", deploymentName, err)
+	}
+	count := scale.Spec.Replicas
+	scale.ResourceVersion = "" // indicate the scale update should be unconditional
+	scale.Spec.Replicas = 0
+	err = waitForDeploymentUpdateScale(c, ns, deploymentName, scale, timeout)
+	if err != nil {
+		return err
+	}
+
+	// Update deployment.
+	deployment, err := c.AppsV1().Deployments(ns).Get(context.TODO(), deploymentName, metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("error get deployment %s: %w", deploymentName, err)
+	}
+	containers := deployment.Spec.Template.Spec.Containers
+	for i := range containers {
+		if containers[i].Name == "csi-rbdplugin" || containers[i].Name == "csi-rbdplugin-controller" {
+			containers[i].Args = append(containers[i].Args, fmt.Sprintf("--clustername=%s", defaultClusterName))
+			deployment.Spec.Template.Spec.Containers[i].Args = containers[i].Args
+		}
+	}
+	// clear creationTimestamp, generation, resourceVersion, and uid
+	deployment.CreationTimestamp = metav1.Time{}
+	deployment.Generation = 0
+	deployment.ResourceVersion = "0"
+	deployment.UID = ""
```

Comment:
@Madhu-1 I hope this is what you mean, right?

Reply:
@Madhu-1 even with the suggested changes the cluster-name is not getting updated in the deployment.
```
May 16 09:07:17.915: FAIL: failed to get cluster name --pool=replicapool/csi-snap-2aa9c5ee-d4ef-11ec-9909-de692c557005 csi.ceph.com/cluster/name: err=command terminated with exit code 2 stdErr="failed to get metadata csi.ceph.com/cluster/name of image : (2) No such file or directory\nrbd: getting metadata failed: (2) No such file or directory\n"
```

Reply:
I need to check on that, but how come this is working for helm and not working for non-helm?

Reply:
May be spurious!!